src/entities/Proposal/jobs.ts

```diff
@@ -236,3 +202,48 @@ export async function publishBids(context: JobContext) {
     ErrorService.report('Error publishing bids', { error, category: ErrorCategory.Job })
   }
 }
+
+const pool = new Pool({ connectionString: process.env.CONNECTION_STRING })
+async function updateProposalsAndBudgets(proposalsWithOutcome: ProposalWithOutcome[], budgetsWithUpdates: Budget[]) {
+  if (proposalsWithOutcome.length === 0) return
+  const client = await pool.connect()
+
+  try {
+    await client.query('BEGIN')
+
+    const proposalUpdateQueriesByStatus: SQLStatement[] = []
+    Object.values(ProposalStatus).forEach((proposalStatus) => {
+      const proposalsToUpdate = proposalsWithOutcome.filter((proposal) => proposal.newStatus === proposalStatus)
+      if (proposalsToUpdate.length > 0) {
+        const query = ProposalModel.getFinishProposalQuery(
+          proposalsToUpdate.map(({ id }) => id),
+          proposalStatus
+        )
+
+        if (query !== null) {
+          proposalUpdateQueriesByStatus.push(query)
+        }
+      }
+    })
+
+    const budgetUpdateQueries = BudgetService.getBudgetUpdateQueries(budgetsWithUpdates)
```

Comment:
Shouldn't we skip this if `budgetsWithUpdates.length === 0`?

Reply:
We could check if `budgetsWithUpdates` has any difference with `currentBudgets`